crypto/interface.go

```diff
@@ -73,12 +73,15 @@ type PublicKeyStore interface {
 	RevokePublicKey(kid string, validTo time.Time) error
 }
 
-// PrivateKeyStore defines the functions for working with private keys.
-type PrivateKeyStore interface {
+type PrivateKeyChecker interface {
```

Comment:
Not entirely sure whether this interface adds something or just increases complexity?

Reply:
Single responsibility. I dont need the jungle and the monkey, just the banana: checking the existence of a private key, so the PrivateKeyStore is clearly way to convoluted with a keyCreator and even several JWS signing methods.